src/mam/mod_mam_utils.erl

```diff
@@ -323,13 +324,19 @@ is_archivable_message(Mod, Dir, Packet=#xmlel{name = <<"message">>}, ArchiveChat
 is_archivable_message(_, _, _, _) ->
     false.
 
+is_valid_message_type(mod_inbox, _, <<"groupchat">>) -> true;
 is_valid_message_type(_, _, <<"normal">>) -> true;
 is_valid_message_type(_, _, <<"chat">>) -> true;
 is_valid_message_type(_, incoming, <<"groupchat">>) -> true;
 is_valid_message_type(_, _, <<"error">>) -> false;
 is_valid_message_type(_, _, _) -> false.
 
 is_valid_message(_Mod, _Dir, Packet, ArchiveChatMarkers) ->
+    %% Message Processing Hints (XEP-0334)
+    Store = exml_query:path(Packet, [{element_with_ns, <<"store">>, ?NS_HINTS}], false) =/= false,
+    Store orelse is_valid_message(Packet, ArchiveChatMarkers).
```

Comment:
Why not add `store` to the `has_any` list in line 351 below to stay consistent with the check for `no-store`?

Reply:
Because store means always store, regardless of everything else, so I thought of saving the computation to check all the other stuff and simply say yes to that one.

Reply:
fixed